python/mxnet/runtime.py

```diff
@@ -69,15 +69,21 @@ def feature_list():
     features = [lib_features_c_array[i] for i in range(lib_features_size.value)]
     return features
 
+
 class Features(collections.OrderedDict):
     """
     OrderedDict of name to Feature
     """
+    feature_dict = collections.OrderedDict()
+    cached = False
     def __init__(self):
-        super(Features, self).__init__([(f.name, f) for f in feature_list()])
+        if not Features.cached:
```

Comment:
Can't you just check if the dictionary is empty and save the bool?

Reply:
Can you better convert it to a singleton which would be more idiomatic and clear to readers?


https://python-3-patterns-idioms-test.readthedocs.io/en/latest/Singleton.html